### DISCUSSION TITLE
Strange behavior of boot.ipxe/wimboot

### DISCUSSION THREAD

**Discussion by madurani** [ipxe/ipxe#1114] [Q&A]
I configured EFI boot via dnsmasq and boot.ipxe. At first glance all look like, that work fine, but installation of win11 not work properly. Some time it work and sometime no.

Description of issue: 
After booting of laptop via uefi/ipx boot.ipxe is downloaded to laptop. Mentioned file automatically downloaded file:  autoexec.ipxe
```
#!ipxe
dhcp
chain http://192.168.1.1/win11pe/boot.ipxe
```
this file downloaded next file boot.ipxe

```
#!ipxe

set server_ip 192.168.1.1  

kernel http://${server_ip}/win11pe/wimboot
initrd -n install.bat http://${server_ip}/win11pe/install.bat
initrd -n winpeshl.ini http://${server_ip}/win11pe/winpeshl.ini
initrd -n BCD http://${server_ip}/win11pe/BCD
initrd -n boot.sdi http://${server_ip}/win11pe/boot.sdi
initrd -n boot.wim http://${server_ip}/win11pe/boot.wim
```
there is install script install.bat
```
wpeinit

reg add HKLM\SYSTEM\Setup\LabConfig
REG ADD HKLM\SYSTEM\Setup\LabConfig /v BypassTPMCheck /t REG_DWORD /d 1
REG ADD HKLM\SYSTEM\Setup\LabConfig /v BypassRAMCheck /t REG_DWORD /d 1
REG ADD HKLM\SYSTEM\Setup\LabConfig /v BypassSecureBootCheck /t REG_DWORD /d 1
net use Z: \\192.168.1.1\win11sk /user:install
Z:
setup.exe /unattend:\\192.168.1.1\install11\tml1127.xml
pause
```
Problem appear in last steps of script, where should be mounted shares of samba.
Attempt mount of shares very often finish with error, that they not exist.
"The network path was not found"
![20240101_174932](https://github.com/ipxe/ipxe/assets/49711883/ededaf69-035b-45e9-a297-f3f0a1416e1d)

 At times help when i turn off installed laptop and restart dnsmasq,samba and httpd, but sometime no. When is laptop long time turn off then all work properly. When I am trying more times boot/install same laptop problem is still back. 
Shares of samba on other laptops where windows is running work without problem. 

Attempt of installation win11 after a few minutes ago. Was only samba,httpd and dnsmasq restarted and installation work fine:

![20240101_175716](https://github.com/ipxe/ipxe/assets/49711883/76ff7d7b-4faa-42b9-bac3-4d19fcb4d26b)

![20240101_175835](https://github.com/ipxe/ipxe/assets/49711883/62ce65b9-0859-48e9-a812-48a2ff0d9c7c)

Samba share:
```
[win11sk]
browsable = true
read only = yes
guest ok = yes
#valid users = install
path = /volume01/http/mediasets/Win11_22H2_Slovak_x64v2

[install11]
browseable = yes
read only = yes
guest ok = yes
#valid users = install
path = /volume01/http/win_tools
```

**Comment by NiKiZe** [comment]
You no longer need BCD or boot.sdi, so you can drop that from your script.

Win10 does no longer allow for anonymous/guest shares, not Sure how it is allowed in some cases for, you but I would start with a secure setup, and then look at the samba logs. More an issue of win10 and samba rather than wimboot or iPXE.

The message indicates that the share isn't available, again since you are using samba, check the logs.

**Reply by madurani** [reply]
I tried comment  BCD and boot.sdi in script, but during of boot i got some error related with BCD.

" anonymous/guest "
I tested shares with user/password access but there was same problem

When problem appear in samba logs aren't any records.

**Reply by NiKiZe** [reply]
Use newer version of wimboot.

If no logs consider IP issue, verify with tcpdump/wireshark or other packet capture tool.

Also try with a "stable client" to unmount and mount

**Reply by madurani** [reply]
wimboot i downloaded from: https://github.com/ipxe/wimboot/releases/latest/download/wimboot

**Comment by epacua** [comment]
I would first check if public share is accessible from another machine. Perhaps try multiple times. Then I would drop at WinPE shell and check if public samba is accessible there. Then check for presence of both winpeshl.ini and install.cmd at %SYSTEMROOT%\System32 which most likely translates to X:\Windows\System32.

With samba, not sure if public access require a username because you have it on your script?
